animation_nodes/data_structures/meshes/mesh_data.pyx

```diff
@@ -370,3 +384,272 @@ def calculateLinkedVertices(int verticesAmount, EdgeIndicesList edges):
+            polyVertices = polyVertices.reversed()
+
+        for j in range(polyLength):
+            amount = indices.length
+
+            if amount == 3:
+                _newPolyStarts[triIndex] = polyIndex
+                _newPolyLengths[triIndex] = 3
+                triIndex += 1
+
+                _newIndices[polyIndex] = neighbors.data[indices.data[0]]
+                _newIndices[polyIndex + 1] = neighbors.data[indices.data[1]]
+                _newIndices[polyIndex + 2] = neighbors.data[indices.data[2]]
+                polyIndex += 3
+
+                break
+
+            # Calculate inner-angle for all vertices of polygon
+            angles = LongList(length = amount)
+            for k in range(amount):
+                preVertex = polyVertices.data[indices.data[(amount + k - 1) % amount]]
+                earVertex = polyVertices.data[indices.data[k]]
+                nexVertex = polyVertices.data[indices.data[(k + 1) % amount]]
+
+                subVec3(&ab, &earVertex, &preVertex)
+                subVec3(&bc, &earVertex, &nexVertex)
+
+                normalizeVec3_InPlace(&ab)
+                normalizeVec3_InPlace(&bc)
+
+                angle = angleNormalizedVec3(&ab, &bc)
+                angles.data[k] = int((angle - angle % 0.001) * 180.0 / pi)
+            sortAngles = LongList.fromNumpyArray(numpy.sort(angles.asMemoryView()).astype(int))
+
+            # Removing an ear which has smallest inner-angle
+            for k in range(amount):
+                index = angles.index(sortAngles[k])
+                preIndex = indices.data[(amount + index - 1) % amount]
+                earIndex = indices.data[index]
+                nexIndex = indices.data[(index + 1) % amount]
+
+                v1 = polyVertices.data[preIndex]
+                v2 = polyVertices.data[earIndex]
+                v3 = polyVertices.data[nexIndex]
+                if convexVertex(v1, v2, v3) and not pointsInTriangle(polyVertices, v1, v2, v3, preIndex, earIndex, nexIndex):
+                    indices = LongList.fromNumpyArray(numpy.delete(indices.asMemoryView(), index).astype(int))
+
+                    _newPolyStarts[triIndex] = polyIndex
+                    _newPolyLengths[triIndex] = 3
+                    triIndex += 1
+
+                    _newIndices[polyIndex] = neighbors.data[preIndex]
+                    _newIndices[polyIndex + 1] = neighbors.data[earIndex]
+                    _newIndices[polyIndex + 2] = neighbors.data[nexIndex]
+                    polyIndex += 3
+
+                    break
+
+    return newPolygons
+
+# Make sure normal (+z) are correct.
+@cython.cdivision(True)
+cdef polyCounterClockwise(Vector3DList vertices):
+    cdef Py_ssize_t amount = vertices.length
+    cdef float area = 0
+    cdef Vector3 v1, v2
+    cdef Py_ssize_t i
+
+    for i in range(amount):
+        v1 = vertices.data[i]
+        v2 = vertices.data[(i + 1) % amount]
+        area += (v2.x - v1.x) * (v2.y + v1.y)
```

Comment:
Instead of using modulo here. We can pull the last case out like this:

```python
    for i in range(amount - 1):
        v1 = vertices.data[i]
        v2 = vertices.data[i + 1]
        area += (v2.x - v1.x) * (v2.y + v1.y)
    v1 = vertices.data[amount - 1]
    v2 = vertices.data[0]
    area += (v2.x - v1.x) * (v2.y + v1.y)
```

Reply:
Done.